### ISSUE TITLE
Pillar keys overwriting in the environment configuration

### ISSUE THREAD

**Issue by ftoek** [saltstack/salt#45082]
The same pillar keys don't overwrite if host is in other environment.

## Configuration:

master:
```
pillar_roots:
  base:
    - /opt/salt-config/pillar/base
  try:
   - /opt/salt-config/pillar/try
   - /opt/salt-config/pillar/base
```

pillar top.sls:
```
base:
  '*':
  - global

try:
  'host*':
  - global-try
```

base/global.sls:
```
VAR: test
```

try/global-try.sls:
```
VAR: 1
```

## EXPECTED RESULTS:

Command 
```
salt host pillar.get VAR
```
returns "1".

## ACTUAL RESULTS:

Command 
```
salt host pillar.get VAR
```
returns "test".

versions-report:
```
Salt Version:
           Salt: 2016.11.6

Dependency Versions:
           cffi: 1.10.0
       cherrypy: Not Installed
       dateutil: Not Installed
      docker-py: Not Installed
          gitdb: Not Installed
      gitpython: Not Installed
          ioflo: Not Installed
         Jinja2: 2.9.6
        libgit2: Not Installed
        libnacl: Not Installed
       M2Crypto: 0.26.0
           Mako: Not Installed
   msgpack-pure: Not Installed
 msgpack-python: 0.4.8
   mysql-python: Not Installed
      pycparser: 2.18
       pycrypto: 2.6.1
   pycryptodome: Not Installed
         pygit2: Not Installed
         Python: 2.6.6 (r266:84292, Nov 21 2013, 12:39:37)
   python-gnupg: Not Installed
         PyYAML: 3.12
          PyZMQ: 16.0.2
           RAET: Not Installed
          smmap: Not Installed
        timelib: Not Installed
        Tornado: 4.3
            ZMQ: 4.0.5

System Versions:
           dist: redhat 6.4 Carbon
        machine: x86_64
        release: 2.6.32-042stab108.2
         system: Linux
        version: Scientific Linux 6.4 Carbon
```

**Comment by garethgreenaway** [comment]
@ftoek I believe you're seeing the results you're seeing because of the ordering of the files in your top.sls file.  Because the base environment file is last that's the one that is taking effect.

**Comment by ftoek** [comment]
Changed top.sls in this way
```
try:
  'host*':
  - global-try

base:
  '*':
  - global
```
nothing changed, the key from global-try had been overwriting as the last time. Also I tried to change order of directory viewing in `pillar_roots:` in `master` it doesn't work and it was expected because in exactly this way pillars work according to [Pillar Namespace Flattening](https://docs.saltstack.com/en/latest/topics/pillar/#pillar-dictionary-merging).
In my case I'd like to have keys in `try` not overwritten from `base`. Now I have to have all hosts in `base`.
